### PR TITLE
[NoQA] [3rd Party Feeds] Workspace feeds clean up

### PR DESCRIPTION
**Pull request by VickyStash**
### Details

Workspace feeds clean up, and API calls integration.

### Fixed Issues

$ https://github.com/Expensify/App/issues/49690
PROPOSAL: N/A

### Tests

- [x] Verify that no errors appear in the JS console

1. Login the app with an account that has `COMPANY_CARD_FEEDS` beta enabled (❗ but not `DIRECT_FEEDS` beta).
2. Create a workspace and enable **Company cards** feature.
3. Go to Company cards page.
4. Tap `Add company cards` button and go through the new feed adding flow. After you finish you will see pending state of the feed.
5. Go to the feed setting and check that `Card feed name update` and `Allow deleting transactions` options work as expected. (❗ The `Remove feed` option won't work since API is in progress.)
6. To test more functionality you need an active feed, to get it please contact me or @robertjchen.
7. In the active feed go through `Assign card` flow and make sure the card is assigned.
8. Open card details and make sure it looks ok. 
9. Try to `Unassign card` and make sure it works as expected. (❗ `Update card` is not working on the API side yet).
10. Assign another card to your user.
11. Go to the workspace members screen and open your member details. Make sure the assigned card is visible there.

### Offline tests

Same as in the **Tests** section.

### QA Steps

- [x] Verify that no errors appear in the JS console

Same as in the **Tests** section.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/ec51e16b-2eeb-48fa-a18e-d01bf091f6c1

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/46d64585-8a88-45f9-8759-7c54667756ad

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/9b2d1f7c-4a59-4646-aa89-62e6280f33e4

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/a12c8cbc-e7db-4f99-b13c-890ab7743661

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/e923029a-7cb4-4bbc-9949-5b494ad23f96

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/e1426fba-a002-4229-b816-6be9bd57e16d

</details>

### CODE REVIEW

**Comment by VickyStash** [comment]
Known API related issues:

1. `CARDASSIGNED` report action display => message in the chat about assigned card is not displayed.
Reason: **OpenReport** API call returns `CARDASSIGNED` report action with `message: []` , which makes it be hidden in the chat.
 Should be fixed by API (slack [link](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1728469195187489?thread_ts=1727729604.987049&cid=C06ML6X0W9L)).
2. **Remove Feed - API call in under work.**
3. **SyncCard - API call returns 404 error**
4. @robertjchen Can we add `lastScrape` and `scrapeMinDate` in cards under cards_WORKSPACE_ID_FEED key:
image
It should make the FE more clean and consistent.
5. If there is no feeds `OpenPolicyCompanyCardsPage` returns empty array as `sharedNVP_private_domain_member_` key
image
Which leads  to the error:
image

**Comment by VickyStash** [comment]
There is an issue on **ios** which can affect the testing
Simulator Screenshot - iPhone 16 - 2024-10-15 at 10 04 57

It's the same on the main, and also affects the Expensify Card screen:
Simulator Screenshot - iPhone 16 - 2024-10-15 at 10 05 03

@allgandalf I remember you mentioned it. Have we created an issue for that?

**Comment by VickyStash** [comment]
@allgandalf Any ETA on reviewing this one?

**Comment by VickyStash** [comment]
@allgandalf You can try to login in my account, just ping me in slack.
Or maybe @robertjchen can add a test setup to your account.

**Comment by VickyStash** [comment]
Recording after conflict resolution:

https://github.com/user-attachments/assets/c7b4c809-1b5c-4621-9d99-ee559f3279b0

**Review by allgandalf — APPROVED** [review]
Looks neat, Code changes look good to me, up for your review @mountiny

File: src/libs/actions/Policy/Policy.ts [hunk 1]
```diff
@@ -4421,6 +4435,23 @@ function enablePolicyAutoReimbursementLimit(policyID: string, enabled: boolean)
     });
 }
 
+function addNewCompanyCardsFeed(policyID: string, feedType: string, feedDetails: string) {
```
Comment (mountiny) [inline src/libs/actions/Policy/Policy.ts:4438]: Why are these in Policy.ts? I think we could have created something like CompanyCards.ts actions file for these actions

**Review by mountiny — APPROVED** [review]
Great job @VickyStash @allgandalf

File: src/libs/actions/Policy/Policy.ts (hunk 1 above)
Comment (VickyStash) [reply src/libs/actions/Policy/Policy.ts:4438]: I can prepare a follow-up and move all of the company cards related methods there!

Comment (VickyStash) [reply src/libs/actions/Policy/Policy.ts:4438]: Follow uo PR is ready: https://github.com/Expensify/App/pull/51080
cc @allgandalf

**Comment by mvtglobally** [comment]
@mountiny Do we need to QA this PR? Can you pls help with more QA friendly steps? Seems like current is more backend heavy checks. 
Also, Can we use Expensifail account for this?

**Comment by mountiny** [comment]
@mvtglobally sorry yeah I agree the tests are more technical. I have marked this as NoQA as testing the custom feeds will be done internally soon in production

File: src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx [hunk 2]
```diff
@@ -61,8 +61,8 @@ function WorkspaceMemberDetailsPage({personalDetails, policy, route}: WorkspaceM
     const {translate} = useLocalize();
     const StyleUtils = useStyleUtils();
     const currentUserPersonalDetails = useCurrentUserPersonalDetails();
-    const [expensifyCardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${workspaceAccountID}_${CONST.EXPENSIFY_CARD.BANK}`);
-    const [allCardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}`);
+    const [allCardsList] = useOnyx(`${ONYXKEYS.CARD_LIST}`);
```
Comment (koko57) [inline src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx:64]: @VickyStash why was this change necessary?
```
    const [allCardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}`);
    const [allCardsList] = useOnyx(`${ONYXKEYS.CARD_LIST}`);
```

Comment (VickyStash) [reply src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx:64]: @koko57 On the members details page, we need to show not only expensify cards, but also custom feeds/direct feeds cards.
They are not stored in the `cards_` right away, we need to do API calls for every feed type to get the data, which can be a pretty big amount of calls.

So I've [asked](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1728482728126749?thread_ts=1727729604.987049&cid=C06ML6X0W9L) @robertjchen about the way I can get all user-related cards at once. And he [suggested](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1728544957562749?thread_ts=1727729604.987049&cid=C06ML6X0W9L) going with `cardList_` key.

So that's the reason of the update.

File: src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx [hunk 3]
```diff
@@ -79,17 +79,18 @@ function WorkspaceMemberDetailsPage({personalDetails, policy, route}: WorkspaceM
...
+    }, [policyID, workspaceAccountID]);
 
     const memberCards = useMemo(() => {
-        if (!expensifyCardsList) {
+        if (!allCardsList) {
             return [];
         }
-        return Object.values(expensifyCardsList).filter((expensifyCard) => expensifyCard.accountID === accountID);
-    }, [expensifyCardsList, accountID]);
+        return Object.values(allCardsList ?? {}).filter((card) => card.accountID === accountID && workspaceAccountID.toString() === card.fundID);
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx:92]: cc @allgandalf

[the remark this answers is not in the document]
Comment (mountiny) [reply src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx:92]: I guess this condition ensures that we only see the cards assigned on this policy in the workspace member page, which is expected

[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx:92]: @mountiny Could you ensure that the BE always returns fundID on cards_ fields (expensify cards)

[the remark this answers is not in the document]
Comment (mountiny) [reply src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx:92]: Well the fundID is the id in the  key so its (cards_fundID_Expensify Card so its kinda duplication

[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx:92]: I just tried again with a new account and fundID has appeared. It seems this only happen with cards that is created long time ago

File: src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx [hunk 4]
```diff
@@ -143,26 +144,26 @@ function WorkspaceCompanyCardDetailsPage({route}: WorkspaceCompanyCardDetailsPag
                                     />
                                 }
                                 description={translate('workspace.moreFeatures.companyCards.lastUpdated')}
-                                title={card?.isLoadingLastUpdated ? translate('workspace.moreFeatures.companyCards.updating') : card?.lastUpdated}
+                                title={card?.isLoadingLastUpdated ? translate('workspace.moreFeatures.companyCards.updating') : card?.lastScrape}
```
Comment (DylanDylann) [inline src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx:147]: we need tp update lastScrape time to the correct timezone: https://github.com/Expensify/App/issues/55880

Comment (mountiny) [reply src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx:147]: interesting

Comment (VickyStash) [reply src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx:147]: I guess one of the reasons why we missed it during the testing is cause the `SyncCard` call didn't work at that moment (see one of my [comments in the slack](https://expensify.slack.com/archives/C06ML6X0W9L/p1729182906054329?thread_ts=1728941405.662029&cid=C06ML6X0W9L)), so maybe we didn't pay it a lot of attention 😞

File: src/pages/workspace/companyCards/WorkspaceCompanyCardEditCardNamePage.tsx [hunk 5]
```diff
@@ -27,16 +27,15 @@ type WorkspaceCompanyCardEditCardNamePageProps = StackScreenProps<SettingsNaviga
 function WorkspaceCompanyCardEditCardNamePage({route}: WorkspaceCompanyCardEditCardNamePageProps) {
     const {policyID, cardID, bank} = route.params;
     const workspaceAccountID = PolicyUtils.getWorkspaceAccountID(policyID);
+    const [customCardNames] = useOnyx(ONYXKEYS.NVP_EXPENSIFY_COMPANY_CARDS_CUSTOM_NAMES);
+    const defaultValue = customCardNames?.[cardID];
```
Comment (ikevin127) [inline src/pages/workspace/companyCards/WorkspaceCompanyCardEditCardNamePage.tsx:31]: Looks like this logic wasn't enough to display the value on mount since `useOnyx` is not loaded instantly and input `defaultValue` is immutable after initial mount, leading to this issue:
- https://github.com/Expensify/App/issues/62076

which we fixed by implementing an action function that's using `Onyx.connect` outside of the component's lifecycle and added it as fallback here to `defaultValue`, this way we'll have the value available on mount.

File: src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx [hunk 6]
```diff
@@ -302,35 +300,23 @@ function WorkspaceMemberDetailsPage({personalDetails, policy, route}: WorkspaceM
                                             </View>
                                             {memberCards.map((memberCard) => (
                                                 <MenuItem
-                                                    title={memberCard.nameValuePairs?.cardTitle}
-                                                    badgeText={CurrencyUtils.convertToDisplayString(memberCard.nameValuePairs?.unapprovedExpenseLimit)}
-                                                    icon={ExpensifyCardImage}
+                                                    key={memberCard.cardID}
+                                                    title={memberCard.nameValuePairs?.cardTitle ?? memberCard?.cardName}
```
Comment (hoangzinh) [inline src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx:304]: Coming from https://github.com/Expensify/App/issues/62214, When we introduce NVP_EXPENSIFY_COMPANY_CARDS_CUSTOM_NAMES, we need to fallback value here as well. You can check more RCA [here](https://github.com/Expensify/App/issues/62214#issuecomment-2893302336)

File: src/libs/ReportActionsUtils.ts [hunk 7]
```diff
@@ -1742,9 +1748,12 @@ function getCardIssuedMessage(reportAction: OnyxEntry<ReportAction>, shouldRende
     const isPolicyAdmin = PolicyUtils.isPolicyAdmin(PolicyUtils.getPolicy(policyID));
     const assignee = shouldRenderHTML ? `<mention-user accountID="${assigneeAccountID}"/>` : assigneeDetails?.firstName ?? assigneeDetails?.login ?? '';
     const navigateRoute = isPolicyAdmin ? ROUTES.EXPENSIFY_CARD_DETAILS.getRoute(policyID, String(cardID)) : ROUTES.SETTINGS_DOMAINCARD_DETAIL.getRoute(String(cardID));
-    const link = shouldRenderHTML
+    const expensifyCardLink = shouldRenderHTML
         ? `<a href='${environmentURL}/${navigateRoute}'>${Localize.translateLocal('cardPage.expensifyCard')}</a>`
         : Localize.translateLocal('cardPage.expensifyCard');
+    const companyCardLink = shouldRenderHTML
```
Comment (dukenv0307) [inline src/libs/ReportActionsUtils.ts:1754]: Coming from: https://github.com/Expensify/App/issues/64264. We should render the companyCardLink if the user is an assignee
